R/pr.R

```diff
@@ -0,0 +1,556 @@
+#' Create a new Plumber router
+#'
+#' @param filters a list of plumber filters
+#' @param file path to file to plumb
+#' @param envir an environment to be used as the enclosure for the routers execution
+#'
+#' @return A new `plumber` router
+#'
+#' @examples
+#' \dontrun{
+#' pr() %>%
+#'   pr_run()
+#' }
+#'
+#' @export
+pr <- function(file = NULL,
+               filters = defaultPlumberFilters,
+               envir = new.env(parent = .GlobalEnv)) {
+  plumber$new(file = file, filters = filters, envir = envir)
+}
+
+#' Add handler to Plumber router
+#'
+#' This collection of functions creates handlers for a Plumber router.
+#'
+#' The generic [pr_handle()] creates a handle for the given methods. Specific
+#' functions are implemented for the following HTTP methods:
+#' * `GET`
+#' * `POST`
+#' * `PUT`
+#' * `DELETE`
+#' * `HEAD`
+#' Each function mutates the plumber router in place, but also invisibly returns
+#' the updated router.
+#'
+#' @param pr A plumber router
+#' @param methods Character vector of HTTP methods
+#' @param path The endpoint path
+#' @param handler a handler function
+#' @param preempt a preempt function
+#' @param serializer a plumber serializer
+#' @param endpoint a `PlumberEndpoint` object
+#' @param ... additional arguments for `PlumberEndpoint`
+#'
+#' @return A plumber router with the handler added
+#'
+#' @examples
+#' \dontrun{
+#' pr() %>%
+#'   pr_handle("GET", "/hi", function() "Hello World") %>%
+#'   pr_run()
+#'
+#' pr() %>%
+#'   pr_handle(c("GET", "POST"), "/hi", function() "Hello World") %>%
+#'   pr_run()
+#'
+#' pr() %>%
+#'   pr_get("/hi", function() "Hello World") %>%
+#'   pr_post("/echo", function(req, res) {
+#'     if (req$postBody == "") return("No input")
+#'     input <- jsonlite::fromJSON(req$POST_BODY)
+#'     list(
+#'       input = input
+#'     )
+#'   }) %>%
+#'   pr_run()
+#' }
+#'
+#' @export
+pr_handle <- function(pr,
+                      methods,
+                      path,
+                      handler,
+                      preempt,
+                      serializer,
+                      endpoint,
+                      ...) {
```

Comment:
Add a check like `if (!inherits(pr, "plumber")) stop("`pr` must be an object of class 'plumber'.")`